src/transforms/filter.js

```diff
@@ -33,10 +37,31 @@ exports.attributes = {
     },
     operation: {
         valType: 'enumerated',
-        values: ['=', '<', '>', 'within', 'notwithin', 'in', 'notin'],
+        values: [].concat(INEQUALITY_OPS).concat(INTERVAL_OPS).concat(SET_OPS),
         dflt: '=',
         description: [
-            'Sets the filter operation.'
+            'Sets the filter operation.',
+
+            '*=* filters items equal to `value`',
+
+            '*<* filters items less than `value`',
+            '*<=* filters items less than or equal to `value`',
+
+            '*>* filters items greater than `value`',
+            '*>=* filters items greater than or equal to `value`',
+
+            '*[]* filters items inside `value[0]` to value[1]` including both bounds`',
+            '*()* filters items inside `value[0]` to value[1]` excluding both bounds`',
+            '*[)* filters items inside `value[0]` to value[1]` including `value[0]` but excluding `value[1]',
+            '*(]* filters items inside `value[0]` to value[1]` including both bounds`',
+
+            '*][* filters items outside `value[0]` to value[1]` and not equal to both bounds`',
+            '*)(* filters items outside `value[0]` to value[1]`',
+            '*](* filters items outside `value[0]` to value[1]` and not equal to `value[0]`',
+            '*)[* filters items outside `value[0]` to value[1]` and not equal to `value[1]`',
+
+            '*{}* filters items present in a set of values',
+            '*}{* filters items not present in a set of values'
```

Comment:
good point - done in https://github.com/plotly/plotly.js/pull/978/commits/7cfa4ffaf0ae51e14fda6d64229e7d1f18bd09a6